spyder/api/asyncdispatcher.py

```diff
@@ -43,12 +57,18 @@ async def my_coroutine():
     ```
     """
 
+    __rlock = threading.RLock()
+
     __closed = False
-    __running_loops: typing.ClassVar[dict[int, asyncio.AbstractEventLoop]] = {}
-    __running_threads: typing.ClassVar[dict[int, threading.Thread]] = {}
+    __running_threads: typing.ClassVar[dict[typing.Hashable, _LoopRunner]] = {}
     _running_tasks: typing.ClassVar[list[Future]] = []
 
-    def __init__(self, coro, *, loop=None, early_return=True):
+    def __init__(self,
+                 async_func: typing.Callable[..., typing.Coroutine[
+                             typing.Any, typing.Any, typing.Any]],
+                 *,
+                 loop: LoopID | None = None,
+                 early_return: bool = True):
```

Comment:
```suggestion
    def __init__(
        self,
        async_func: typing.Callable[
            ..., typing.Coroutine[typing.Any, typing.Any, typing.Any]
        ],
        *,
        loop: LoopID | None = None,
        early_return: bool = True
    ):
```

Formatting as suggested by Black.